src/main/java/net/sf/jabref/model/database/BibDatabaseContext.java

```diff
@@ -176,6 +176,7 @@ public boolean isBiblatexMode() {
     * @return The default directory for this field type.
     */
     public List<String> getFileDirectories(String fieldName, FileDirectoryPreferences preferences) {
+        // TODO: all of these folder will be also searched for local PDFs. If this includes large folders like user.home this might take forever.
```

Comment:
The findAssociated files gets a list of the dirs but uses a lazy stream (Files.find) to populate a HashSet.
Should not be that performance problem.
However, I see the findAssociatedFiles method could be improved to directly search on the stream. But this is time for a new PR